barretenberg/cpp/pil/avm/avm_main.pil

```diff
@@ -2,8 +2,22 @@
 include "avm_mem.pil";
 include "avm_alu.pil";
 include "avm_binary.pil";
+include "avm_environment.pil";
 
 namespace avm_main(256);
+    //===== PUBLIC INPUTS =========================================================
+    pol public kernel_inputs;
+
+    // Envrionment lookup selector opcodes
+    // NOTE: we start with an arbitrary subset of operations - just these two for now
+    // CALLER
+    pol commit sel_op_sender;
+    // ADDRESS
+    pol commit sel_op_address;
+    
+    #[LOOKUP_INTO_ENVIRONMENT]
+    // TODO: FIX not having the trailing is_public breaking compilation :(
+    avm_environment.q_environment_lookup { ia, avm_environment.environment_selector } in avm_environment.q_environment_lookup { kernel_inputs__is_public, clk };
```

Comment:
I am surprised it is working with the same selector on both sides. In general, this should not work.

I suggest a re-design, where we introduce a new trace/namespace for public inputs containing the column kernel_inputs and a selector for the lookup (right hand side): avm_public.kernel_sel, avm_public.kernel_inputs

In the main trace, we map any relevant selector to an index:

sel_op_sender * env_idx = 0;
sel_op_addr * (env_idx - 1) = 0;
.....

We add a selector for the lookup (LHS): avm_main.kernel_sel (different column as the one above)
We enforce this selector to be toggled whenever we have an environment operation through the following relation:

( sel_op_sender + sel_op_address + ....) * (1-kernel_sel) = 0; (main trace)

Finally, we apply the lookup from main trace to avm_public:

avm_main.kernel_sel { ia, env_idx } = avm_public.kernel_sel { avm_public.kernel_inputs, clk}